Implemented to tick for each strategy/selector combination after succuessful backfilling

Shawn8901 opened the pull request:
Added a tick-call for every strategy / exchange combination
Added package-lock to gitignore
fixed dist gitignore
renamed zbTypes to m8bTypes
increased engine declaration to require node 10, like required in index.ts

Shawn8901 committed:
Added a tick-call for every strategy / exchange combination

Added package-lock to gitignore
fixed dist gitignore
renamed zbTypes to m8bTypes
increased engine declaration to require node 10, like required in index.ts

Shawn8901 committed:
Load days to backfill from config

notVitaliy commented on package.json:
```diff
@@ -1,6 +1,6 @@
 {
-  "name": "magic8bot4",
-  "version": "5.0.0-alpha-1",
+  "name": "magic8bot",
```
Good catch 👍

notVitaliy commented on .gitignore:
```diff
@@ -3,4 +3,5 @@ node_modules
 .env
-./dist
+dist
+package-lock.json
```
I didn't put it in the docs yet, but can you use `yarn` instead of `npm i`?

Shawn8901 replied:
i will check out yarn, but you really want to check in the lock files? i personally would also add yarn.lock to gitignore. Afterwards everybody could feel free to use yarn or npm.

PS: Nice hit with yarn, i like it!

notVitaliy replied:
The point of the lock file is so that everyone has consistent builds with the same package versions.

notVitaliy commented on src/conf.sample.ts:
```diff
@@ -4,7 +4,7 @@
  */
 
-import { MongoConf, Base, ExchangeConf, Conf, Magic8bot } from '@zbTypes'
+import { MongoConf, Base, ExchangeConf, Conf, Magic8bot } from '@m8bTypes'
```
👍

notVitaliy commented on src/engine/engine.ts:
```diff
@@ -38,15 +49,17 @@ export class Engine {
+    this.backfillers.forEach(async (backfillState, selector) => {
+      backfillState.status = BACKFILL_STATUS.RUNNING
+      await this.backfill(selector, backfillState.days)
+      backfillState.status = BACKFILL_STATUS.DONE
```
Not sure if there's an explicit need for this since `this.backfill` will only return once it's done. It'd be safe to invoke `forEach (strat) -> this.tick` right after.

Shawn8901 replied:
you mean to drop the state? i am okay with that, i just added it because i found your enum.

But with the backfiller state, you could trace it p.e. for ui  if the backfiller is still in running state.

I will update my pr after receiving feedback on that.

notVitaliy replied:
It was a half-finished thought. Just don't really see a use case for it right now.

The plan for exposing data to the UI would be a singleton class. The service that can be imported where needed and then pipe data into it. The service would have some helper methods and hooks.

It'd mostly be a wrapper around a WS server.

notVitaliy reviewed (changes requested):
Meant to do changes request

Shawn8901 committed:
period.store.ts:  fixed issue that High was calculated with Math.min
                  removed copy of old volume, since its not valid for next period.store.ts
session.store.ts: removed magic conversion from date to number
trade.store.ts:   added selector to stored trades
                  merged index of selector and time, because it will be searched in pair anyway
engine.ts:        removed backfillstate and backfill_status

notVitaliy commented on src/store/period.store.ts:
```diff
@@ -30,25 +30,25 @@ export class PeriodStore {
       low: price,
       close: price,
-      volume: size,
+      volume: 0,
```
This should be `size`. Since it's a new trade that came in that's outside the period of the previous period. Otherwise, the first trade from a period won't get its size recorded.

notVitaliy commented on src/store/session.store.ts:
```diff
@@ -3,28 +3,28 @@ import { Collection } from 'mongodb'
 import { mongoService } from '../services/mongo.service'
 
-interface SessionCollection {
+interface SessionItem {
```
Can you revert these 2 store files? I've got a bigger change in the pipe that's going to conflict.

notVitaliy commented on src/store/trade.store.ts:
```diff
@@ -40,10 +39,11 @@ export class TradeStore {
   }
 
   async update(selector: string, newTrades: TradeItem[]) {
+    newTrades.forEach((value) => value['selector'] = selector)
```
Not really a nice way of doing this. You're basically mutating the original object.

```typescript
const tradesCollection = newTrades.map((trade) => ({ ...trade, selector }))
```

Shawn8901 commented:
closing pr since its already included in master branch
